src/lib/forms/core/DateTimeFields/DateInputField.js

```diff
@@ -20,6 +25,15 @@ export class DateInputField extends React.Component {
           handleBlur={props.form.handleBlur}
           handleDateChange={(value, name) => {
             props.onChange(fromISO(value), name);
+            if (_isEmpty(value)) {
```

Comment:
we could change the whole block to be a method of this component, for readability of the code